### PR TITLE
Oraclevs integration

### PR DESCRIPTION
**Pull request by rohanaggarwal7997**
Thank you for contributing to LangChain!

- Oracle AI Vector Search 
  Oracle AI Vector Search is designed for Artificial Intelligence (AI) workloads that allows you to query data based on semantics, rather than keywords. One of the biggest benefit of Oracle AI Vector Search is that semantic search on unstructured data can be combined with relational search on business data in one single system. This is not only powerful but also significantly more effective because you don't need to add a specialized vector database, eliminating the pain of data fragmentation between multiple systems.

- Oracle AI Vector Search is designed for Artificial Intelligence (AI) workloads that allows you to query data based on semantics, rather than keywords. One of the biggest benefit of Oracle AI Vector Search is that semantic search on unstructured data can be combined with relational search on business data in one single system. This is not only powerful but also significantly more effective because you don't need to add a specialized vector database, eliminating the pain of data fragmentation between multiple systems.
This Pull Requests Adds the following functionalities
Oracle AI Vector Search : Vector Store
Oracle AI Vector Search : Document Loader
Oracle AI Vector Search : Document Splitter
Oracle AI Vector Search : Summary
Oracle AI Vector Search : Oracle Embeddings

- We have added unit tests and have our own local unit test suite which verifies all the code is correct. We have made sure to add guides for each of the components and one end to end guide that shows how the entire thing runs. 

- We have made sure that make format and make lint run clean.

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in langchain.

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, hwchase17.

### CODE REVIEW

File: docs/docs/integrations/providers/oracleai.mdx
[the remark this answers is not in the document]
Comment (rohanaggarwal7997) [reply docs/docs/integrations/providers/oracleai.mdx:1]: Yes. The first one is Oracle Generative AI service. Oracle AI vector search uses within database services and supports callbacks to external providers and is set up amazingly well for OCI gen ai.

File: libs/community/langchain_community/document_loaders/oracleai.py [hunk 1]
```diff
@@ -0,0 +1,460 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2023 - , Oracle and/or its affiliates.
+# -----------------------------------------------------------------------------
+# Authors:
+#   Harichandan Roy (hroy)
+#   David Jiang (ddjiang)
+#
+# -----------------------------------------------------------------------------
+# oracleai.py
+# -----------------------------------------------------------------------------
```
Comment (baskaryan) [inline libs/community/langchain_community/document_loaders/oracleai.py:10]: can we remove these copyright statements (if you'd like to contribute this code)? all code in this repo is subject to the main license https://github.com/langchain-ai/langchain/blob/master/LICENSE

Comment (rohanaggarwal7997) [reply libs/community/langchain_community/document_loaders/oracleai.py:10]: Sure

**Comment by rohanaggarwal7997** [comment]
Hi @baskaryan 
We wanted to keep the tests disabled in the langchain repo. We have our own cronjobs at our end to verify the code. What is the easiest way to disable these tests. We do not even provide a username, password and dsn in the tests currently. Right now its failing because its not able to find oracledb driver. Can you please provide some guidance regarding this ?

**Comment by baskaryan** [comment]
> Hi @baskaryan We wanted to keep the tests disabled in the langchain repo. We have our own cronjobs at our end to verify the code. What is the easiest way to disable these tests. We do not even provide a username, password and dsn in the tests currently. Right now its failing because its not able to find oracledb driver. Can you please provide some guidance regarding this ?
> 
> I fixed the lint, spellcheck and copyright error.
> 
> I am not sure why these includes show up as error. These seem to be correct only
> 
> ImportError: Found bad imports: File: /home/runner/work/langchain/langchain/docs/docs/integrations/tools/oracleai.ipynb langchain_community.utilities.oracleai.OracleSummary File: /home/runner/work/langchain/langchain/docs/docs/integrations/vectorstores/oracle.ipynb langchain_community.vectorstores.oraclevs langchain_community.vectorstores.oraclevs.OracleVS File: /home/runner/work/langchain/langchain/docs/docs/integrations/text_embedding/oracleai.ipynb langchain_community.embeddings.oracleai.OracleEmbeddings langchain_community.embeddings.oracleai.OracleEmbeddings File: /home/runner/work/langchain/langchain/docs/docs/integrations/document_loaders/oracleai.ipynb langchain_community.document_loaders.oracleai.OracleDocLoader langchain_community.document_loaders.oracleai.OracleTextSplitter

These should be moved to integration_tests instead of unit_tests

**Comment by rohanaggarwal7997** [comment]
@baskaryan Thanks for the comment. I have moved the tests. 
Can you please run the CI pipeline now ?

**Comment by rohanaggarwal7997** [comment]
Hi @baskaryan can you please approve the ci pipeline ? We want to merge this pull request soon. 
Adding @efriis @eyurtsev in case @baskaryan is busy.

**Comment by shailendrah** [comment]
Hi @baskaryan:

My name is shailendra mishra. I am working with @rohanaggarwal7997 on the langchain integration project. I hope you're doing well. Could you please provide an update on the review process for the feature we are planning to merge to main? Let me know if you have any questions or need further clarifications. Thanks!
CCing: @efriis @eyurtsev - if you have some insight into the status of the merge to main.

**Comment by skmishraoracle** [comment]
Hi hwchase17: (Escalating this to you would greatly appreciate any response)

We are  a group in Oracle responsible for integration of langchain with OracleDBAI (the latest release of Oracle Database). The code is in a branch of langchain-ai/langchain. We requested that after reviews our branch be allowed to merge to main. @baskaryan is our reviewer. We have sent a no. of notes requesting status of the reviews & if we can help him reviewing the code. It appears that he is busy with some other errands. Hence escalating this to you. The following is a note that we send to him:

Hi @baskaryan:
My name is shailendra mishra. I am working with @rohanaggarwal7997 on the langchain integration project. I hope you're doing well. Could you please provide an update on the review process for the feature we are planning to merge to main? Let me know if you have any questions or need further clarifications. Thanks!
CCing: @efriis @eyurtsev - if you have some insight into the status of the merge to main.

**Comment by skmishraoracle** [comment]
Hi @hwchase17: (Escalating this to you would greatly appreciate any response)

We are a group in Oracle responsible for integration of langchain with OracleDBAI (the latest release of Oracle Database). The code is in a branch of langchain-ai/langchain. We requested that after reviews our branch be allowed to merge to main. @baskaryan is our reviewer. We have sent a no. of notes requesting status of the reviews & if we can help him reviewing the code. It appears that he is busy with some other errands. Hence escalating this to you. The following is a note that we send to him:

Hi @baskaryan:
My name is shailendra mishra. I am working with @rohanaggarwal7997 on the langchain integration project. I hope you're doing well. Could you please provide an update on the review process for the feature we are planning to merge to main? Let me know if you have any questions or need further clarifications. Thanks!
CCing: @efriis @eyurtsev - if you have some insight into the status of the merge to main.

**Review by baskaryan — DISMISSED** [review]

**Comment by baskaryan** [comment]
please make sure all imports of optional dependencies like oracledb are conditional (happen inside a function)

**Comment by rohanaggarwal7997** [comment]
> address comments

@baskaryan I have addressed your comments - now the make test on my machine runs fine the way it runs in the langchain CI/CD environment. I think now this PR is good to merge. Can you please take a look. @efriis can you please approve the workflow.

**Review by baskaryan — APPROVED** [review]

**Comment by baskaryan** [comment]
Thank you for the contribution @rohanaggarwal7997 @shailendrah @skmishraoracle! Will be included in the next langchain-community release.

**Comment by shahn33** [comment]
from langchain.vectorstores import OracleVS
ImportError: cannot import name 'OracleVS' from 'langchain.vectorstores'

**Comment by skmishraoracle** [comment]
@shahn33:  Use from langchain_community.vectorstores import OracleVS

**Pull request merged**